modules/wlroots/hm.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+
+with config.stylix.colors;
+
+let
+  slideshow = pkgs.writeShellApplication {
+    name = "slideshow";
+    runtimeInputs = [ pkgs.swww ];
+    text = ''
+      images=(${builtins.toString config.stylix.wallpaper.images})
+      while true; do
+        swww img ''${images[ $RANDOM % ''${#images[@]} ]}
+        sleep ${builtins.toString config.stylix.wallpaper.delay}
```

Comment:
That sounds like a very good idea. However, I'm not sure why images are being selected at random here. Would users expect the slideshow to follow the order they choose?

Reply:
> However, I'm not sure why images are being selected at random here. Would users expect the slideshow to follow the order they choose?

Sounds like this should be a Stylix option.

> Using a timer will only work with randomness, as otherwise we would have no way to remember the index of the current image.

We could cache statefull information in files, like `/tmp/` or `/run`: https://superuser.com/questions/1341811/stop-a-systemd-service-and-clear-all-persistent-data

Reply:
That seems like a lot of extra complication compared to a simple `while` loop.

Reply:
Sadyly, the `while` loop has the same disadvantages as cron jobs compared to Systemd timers.

Particularly, missed triggers due to suspended systems are properly handled with Systemd timers: https://wiki.archlinux.org/title/Systemd/Timers#Realtime_timer

Maybe this edge case is worth implementing a Systemd timer.